instance.go

```diff
@@ -86,8 +88,27 @@ func InstancesFromEC2Result(in *ec2.DescribeInstancesOutput) []*Instance {
 			out = append(out, NewInstance(oi))
 		}
 	}
-	sort.Sort(InstancesByName(out))
-	return out
+
+	var regex *regexp.Regexp
+	if *filterRegex != "" {
+		var err error
+		regex, err = regexp.Compile(*filterRegex)
+		if err != nil {
+			fmt.Println(*filterRegex)
+			log.Fatal("Failed to compile regex %q: %b", *filterRegex)
+		}
+	}
+
+	filteredOut := []*Instance{}
+	for _, i := range out {
+		if *filterRegex != "" && !regex.MatchString(i.Name()) {
+			continue
+		}
+		filteredOut = append(filteredOut, i)
+	}
+
```

Comment:
Looping again just to filter the unwanted instances seems redundant as we're looping over in just above. However, there didn't seem to be an easy way to get the instance name for matching at that point.